firebase-vscode/src/core/emulators.ts

```diff
@@ -241,6 +241,21 @@ export class EmulatorsController implements Disposable {
     };
   }
 
+  public getLocalEndpoint = () => computed<string | undefined>(() => {
+    const emulatorInfos = this.emulators.value.infos?.displayInfo;
+    const dataConnectEmulator = emulatorInfos?.find(
+      (emulatorInfo) => emulatorInfo.name === Emulators.DATACONNECT,
+    );
+
+    if (!dataConnectEmulator) {
+      return undefined;
+    }
+
+    return (
+      "http://" + dataConnectEmulator.host + ":" + dataConnectEmulator.port
```

Comment:
Please handle IPv6 quoting similar to https://github.com/firebase/firebase-tools/blob/1633d690e9ed84f9bfe68455fe76a3135681f0b7/src/emulator/registry.ts#L180-L181